### PR TITLE
Print the number of files that need formatting.

### PR DESCRIPTION
**Pull request by Harry-Hopkinson**
## Description

This change prints the number of files that need to be changed in order to match the Prettier Code Style.

For example if you
```shell
prettier --check <a directory>
```

Now if one file needs to be changed it prints out.
Code style issues found in the above file. Forgot to run Prettier?

Instead of
Code style issues found in the above file(s). Forgot to run Prettier?

Another change is it now prints out the number of files that need to be formatted. For example if you have three files that need to be formatted in a directory it now outputs the number of files that need to change.

```shell
Code style issues found in 3 files. Forgot to run Prettier?
```

This is in contrast to the previous

```shell
Code style issues fixed in the above file(s). Forgot to run Prettier?
```

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
I like the idea, can you add test for multiple unformatted files?

**Comment by Harry-Hopkinson** [comment]
The tests work as expected. It outputs the following...
```shell
Code style issues found in 2 files. Forgot to run Prettier?
```

File: src/cli/format.js [hunk 1]
```diff
@@ -417,11 +417,19 @@ async function formatFiles(context) {
   if (context.argv.check) {
     if (numberOfUnformattedFilesFound === 0) {
       context.logger.log("All matched files use Prettier code style!");
+    } else if (numberOfUnformattedFilesFound === 1) {
+      context.logger.warn(
+        context.argv.write
+          ? "Code style issues fixed in the above file(s)."
```
Comment (fisker) [inline src/cli/format.js:423]:
```suggestion
          ? "Code style issues fixed in the above file."
```

File: src/cli/format.js [hunk 2]
```diff
@@ -417,11 +417,19 @@ async function formatFiles(context) {
...
+          : "Code style issues found in the above file. Forgot to run Prettier?"
+      );
     } else {
       context.logger.warn(
         context.argv.write
           ? "Code style issues fixed in the above file(s)."
```
Comment (fisker) [inline src/cli/format.js:429]:
```suggestion
          ? "Code style issues found in " + numberOfUnformattedFilesFound +" files."
```

**Review by fisker — CHANGES_REQUESTED** [review]

**Comment by fisker** [comment]
Don't merge main into your PR, if it's necessary, we'll ask you to do.

**Comment by Harry-Hopkinson** [comment]
Okay thanks.

**Comment by Harry-Hopkinson** [comment]
Just updating the tests.

**Comment by fisker** [comment]
Can you add changelog for this?

**Comment by Harry-Hopkinson** [comment]
Sure thing

**Comment by Harry-Hopkinson** [comment]
@fisker I have now followed the template.

**Review by fisker — APPROVED** [review]
Thanks! I hope no one uses the stderr directly.

**Comment by Harry-Hopkinson** [comment]
Thank you

**Review by sosukesuzuki — APPROVED** [review]

**Comment by sosukesuzuki** [comment]
Can we merge this? @fisker Do you have other tasks that is should be done in this PR?

**Comment by fisker** [comment]
Thank you for contributing to Prettier! ❤️

**Pull request merged**